core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -66,7 +67,9 @@ static void toJson(ViewMetadata metadata, JsonGenerator gen) throws IOException
     gen.writeStringField(VIEW_UUID, metadata.uuid());
     gen.writeNumberField(FORMAT_VERSION, metadata.formatVersion());
     gen.writeStringField(LOCATION, metadata.location());
-    JsonUtil.writeStringMap(PROPERTIES, metadata.properties(), gen);
+    if (!metadata.properties().isEmpty()) {
```

Comment:
But In case of table metadata we are always writing it. 
https://github.com/apache/iceberg/blob/dd26f3630954e69c584faa06c9395f77d49337b1/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L214

Should we make it similar in a follow up?

Reply:
I think we can probably leave the `TableMetadata` as is. I'm curious why we want to make this change for views, sorry if this was already discussed before @nastra . Why can't we just keep it similar to tables, and have the property always exist, and by default it'll just be the empty collection that Immutables creates?

Reply:
@amogh-jahagirdar: Table metadata , the read is still optional. But always writes. So, else case is a dead code. 
For view this PR is making both read and write optional.

Reply:
@amogh-jahagirdar in newer parsers we typically try to avoid sending empty collections, hence the suggested change.
We could still send an empty collection here for views, but we still need to adjust the read-side to not require properties

Reply:
> We could still send an empty collection here for views, but we still need to adjust the read-side to not require properties

Yeah I'm on board with this @nastra. Checking on the read side is expected since properties is optional as per the spec. I think any Iceberg library parser should write out "richer" metadata even if it's empty. It makes it easy for someone to look at the metadata file and see the state (rather than determine a field is missing). Also minor point in this case, it simplifies the code to just write out the field.

Reply:
Just discussed with @nastra , I also see the other argument that writing empty collections is a bit wasteful. Considering that, I'm okay either way.